### ISSUE TITLE
Svelte types not automatically inferred in `.ts` & `.js` files

### ISSUE THREAD

**Issue by probablykasper** [zed-industries/community#1847]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

In SvelteKit projects, types are automatically inferred for many `export`ed values ([Zero-effort type safety](https://svelte.dev/blog/zero-config-type-safety)).

In `+page.svelte`, it works:
<img width="614" alt="image" src="https://github.com/zed-industries/community/assets/11315492/cb34705f-9d76-4bcb-be5c-df037edcd4d5">

But in `.ts`/`.js` files it does not:
<img width="780" alt="image" src="https://github.com/zed-industries/community/assets/11315492/d3cdda3f-1943-49d9-b7bc-88de82893c11">

### Reproduce
- Run `npm create svelte@latest`
- Choose SvelteKit demo app
- Check `src/routes/sverdle/+page.server.ts` (or `.js`)

### Environment

Zed: v0.96.3 (stable)
OS: macOS 13.5.0
Memory: 32 GiB
Architecture: x86_64

**Comment by mikayla-maki** [comment]
Duplicate of https://github.com/zed-industries/community/issues/1815

**Comment by probablykasper** [comment]
@mikayla-maki I don't think this is a duplicate, unless you mean they have the same root cause. That issue seems to be about types not updating when importing in `.svelte` from a different file, while this issue is about types not being applied in JS/TS files

**Comment by mikayla-maki** [comment]
Fair point, I misread.

**Comment by ChristianPerez34** [comment]
This fixed issues for me [Zed Documentation TypeScript](https://docs.zed.dev/configuration/configuring-zed#typescript)

![image](https://github.com/zed-industries/community/assets/22486123/2b2c82dd-d6b7-4dca-941a-8a394d66603b)
